### PR TITLE
1.0.0-rc1

### PR DESCRIPTION
**Pull request by shefty**
There are still open issues to address before an actual release, but I believe that the interfaces that have been defined and the available providers are in a reasonable enough state to start a more thorough testing process.

Signed-off-by: Sean Hefty <sean.hefty@intel.com>

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -1,7 +1,7 @@
 dnl Process this file with autoconf to produce a configure script.
 
 AC_PREREQ(2.57)
-AC_INIT([libfabric], [0.0.2], [ofiwg@lists.openfabrics.org])
+AC_INIT([libfabric], [1.0.0-rc1], [ofiwg@lists.openfabrics.org])
```
Comment (pmmccorm) [inline configure.ac]: You probably want to get rid of that -rc1 it causes rpmbuild to choke:

```
rpmbuild -ba libfabric.spec
error: line 2: Illegal char '-' in: Version: 1.0.0-rc1
```

File: libfabric.spec.in [hunk 2]
```diff
@@ -1,6 +1,6 @@
 Name: libfabric
 Version: @VERSION@
-Release: 1%{?dist}
+Release: 1.rc1%{?dist}
```
Comment (pmmccorm) [inline libfabric.spec.in]: Also now might be a good time to hard code the configure flags which choose the providers and corresponding rpm buildrequires

File: configure.ac (hunk 1 above)
Comment (shefty) [inline configure.ac]: Can you see if ".rc1" works instead?  We kind of want to mark this as the rc1...

Comment (pmmccorm) [inline configure.ac]: It works: change the version to 1.0.0.rc1 and leave the spec release unchanged (take out rc1), which will then make tarballs how you want.

Comment (jsquyres) [inline configure.ac]: FWIW, I've seen a lot of packages not try to delimit "a" or "b" or "rc" from the version number.  I.e., "1.0.0rc1" (no . or -).  The version number parsers grab the various parts like this:

```perl
    # Typed by hand in github -- not checked!
    $version =~ m/(\d+)\.(\d+)\.(\d+)(.*)/;
    my $major = $1;
    my $minor = $2;
    my $release = $3;
    my $suffix = $4;
```

**Pull request merged**